### PR TITLE
fix: resolve memory leak, by cloning always from the main instance

### LINKED ISSUE #1101: Memory leak caused by cloning the i18n instance
## Describe the bug
The #1073 pr introduced a significant memory leak when using `getServerSideProps`.

## Occurs in next-i18next version
[v8.1.2](https://github.com/isaachinman/next-i18next/releases/tag/v8.1.2)

## Steps to reproduce
1. using [simple example](https://github.com/isaachinman/next-i18next/tree/master/examples/simple) from repo.
2. use `getServerSideProps` instead of `getStaticProps` in [index.js](https://github.com/isaachinman/next-i18next/blob/master/examples/simple/pages/index.js#L42)
3. run `yarn build` on the example repo
4. start server in [debug mode](https://nextjs.org/docs/advanced-features/debugging#step-1-start-nextjs-in-debug-mode). In your `package.json`
```js
"start": "NODE_OPTIONS='--inspect' next start -p ${PORT:=3000}"
```
5. In chrome open `chrome://inspect` url then click on inspect
Screen Shot 2021-03-28 at 2 28 15 PM

6. Tab to the `Memory` tab.
7. Load `localhost:3000`. Take a snapshot (base).
8. Use [loadtest](https://www.npmjs.com/package/loadtest) or any other load testing library to hit localhost:3000 (2000 total requests). Send 1000 requests, stop the load, click collect garbage, take snapshot. Inspect the retained data.
9. Send another 1000 requests, stop the load test, click collect garbage, take a snapshot. Inspect the retained data. NOTE: it will take some time for the snapshot because of the tree created by the leak.
10. Observer the memory significantly increased and the GC never releases it.
11. Inspect the retained objects and you will see the i18n references.

## Expected behaviour
After stopping the loadtest and manually triggering GC (or waiting for it to auto trigger) memory should go back to baseline (plus any additional memory used post first request).
This is the behaviour in [v8.1.1](https://github.com/isaachinman/next-i18next/releases/tag/v8.1.1)

## Screenshots
As you can see the exact same test with the only difference being the version does not show a memory leak.
In 8.1.1 we go from 11.3 -> 12.3
In 8.1.2 we go from 11.3 -> 66.3

**8.1.2**
Screen Shot 2021-03-28 at 2 47 22 PM
Screen Shot 2021-03-28 at 2 42 18 PM
Screen Shot 2021-03-28 at 2 42 08 PM
Screen Shot 2021-03-28 at 2 41 57 PM

**8.1.1**
Screen Shot 2021-03-28 at 2 51 52 PM

## OS (please complete the following information)
 - Device: [MacBook Pro (16-inch, 2019)]
 - Browser: [Chrome Version 89.0.4389.90]

### PR DESCRIPTION
**Pull request by adrai**
fixes #1101 by creating a clone always from the first instance of i18next.

### CODE REVIEW

**Review by dcodus — APPROVED** [review]
Tested:
1. No memory leaks detected ✅ 
2. Stable servers under heavy load ✅ 
3. SSR works as expected for different locales ✅

**Comment by dcodus** [comment]
@isaachinman When do you think we can get this out?

**Comment by adrai** [comment]
@dcodus probably not in the same timezone 😉

File: src/createClient/node.ts [hunk 1]
```diff
@@ -3,30 +3,32 @@ import i18nextFSBackend from 'i18next-fs-backend'
 
 import { InternalConfig, CreateClientReturn, InitPromise, I18n } from '../types'
 
-let instance: I18n
+let mainInstance: I18n
 
 export default (config: InternalConfig): CreateClientReturn => {
-  if (!instance) {
-    instance = i18n.createInstance(config)
+  let currentInstance: I18n
```
Comment (isaachinman) [inline src/createClient/node.ts:9]: Can we maintain the naming of `instance` here, and call `mainInstance` `globalInstance` instead?

**Comment by isaachinman** [comment]
Just a minor comment. Otherwise looks good to me.

File: src/createClient/node.ts (hunk 1 above)
Comment (adrai) [reply src/createClient/node.ts]: done

File: examples/simple/yarn.lock [hunk 2]
```diff
@@ -2485,15 +2485,8 @@ neo-async@^2.5.0, neo-async@^2.6.1, neo-async@^2.6.2:
...
-  version "8.1.0"
-  dependencies:
-    "@types/hoist-non-react-statics" "^3.3.1"
-    "@types/i18next-fs-backend" "^1.0.0"
-    core-js "^3"
-    hoist-non-react-statics "^3.2.0"
-    i18next "^19.8.4"
-    i18next-fs-backend "^1.0.7"
-    react-i18next "^11.8.8"
+  version "0.0.0"
```
Comment (isaachinman) [inline examples/simple/yarn.lock:2488]: Any idea what happened here? This looks unintended to me.

Comment (adrai) [reply examples/simple/yarn.lock]: strange... just reverted the yarn file

**Comment by isaachinman** [comment]
@adrai Can you remove the yarn file change completely?

**Comment by adrai** [comment]
should be gone now

File: src/createClient/node.ts [hunk 3]
```diff
@@ -3,13 +3,15 @@ import i18nextFSBackend from 'i18next-fs-backend'
 
 import { InternalConfig, CreateClientReturn, InitPromise, I18n } from '../types'
 
-let instance: I18n
+let globalInstance: I18n
 
 export default (config: InternalConfig): CreateClientReturn => {
-  if (!instance) {
-    instance = i18n.createInstance(config)
+  let instance: I18n
```
Comment (isaachinman) [inline src/createClient/node.ts:9]:
```suggestion
  let instance: I18n
  
```

File: src/createClient/node.ts [hunk 4]
```diff
@@ -3,13 +3,16 @@ import i18nextFSBackend from 'i18next-fs-backend'
...
+
```
Comment (isaachinman) [inline src/createClient/node.ts:10]:
```suggestion

```

**Pull request merged**